### ISSUE TITLE
Add additional files support

### ISSUE THREAD

**Issue by cs-sid** [sphinx-contrib/tikz#20]
Assume that you have a project where you both want to generate a html page and a pdf (via latex output), and assume your tikz code needs some defined macros. To avoid duplicating the macros in `latex_elements['preamble']` and `tikz_latex_preamble` we write them into `macros.tex` and use `\input{macros}` as preamble. Adding `macros.tex` to `latex_additional_files` will make the latex preamble work, but for the html builder the tikz pictures will not compile due to 
```
! LaTeX Error: File `macros.tex` not found
```
I suggest adding a `tikz_additional_files` configuration option similar to `latex_additional_files`, so that
```
tikz_additional_files = ["macros.tex"]
```
will solve the problem.

**Comment by philexander** [comment]
Thank you for raising this issue.

I think it would make sense to always make all `latex_additional_files` available in the `tikz` code, without introducing another configuration option `tikz_additional_files`. Or do you see a use case for this additional configurability?

**Comment by cs-sid** [comment]
Well, not directly. I think it is more about the fact that there is already a `tikz_latex_preamble` vs a `latex_elements['preamble']`. In theory one could simply use the latex configuration settings also for the tikz extension. (Meaning using `latex_elements['preamble']` instead of `tikz_latex_preamble`, using `latex_additional_files` instead of `tikz_additional_files`, etc).

I guess the most flexible and least intrusive way is to use the `latex_elements['preamble']` and the `latex_additional_files` also in the tikz extension. This way no additional options are needed, and there is less possibility of conflict when using a latex builder. 

For backward compatibility one could use the following approach. Use `latex_elements['preamble']` if no `tikz_latex_preamble` is specified, but the latter if specified. And simply respect `latex_additional_files` as you suggested.

I might be wrong, but as far as I have seen, currently neither `latex_elements['preamble']` nor `latex_additional_files` is respected by the tikz extension.

**Comment by philexander** [comment]
@cs-sid I agree with your reasoning. This behavior is implemented with version 0.4.18.
